src/external/rawspeed/RawSpeed/Common.cpp

```diff
@@ -24,14 +24,20 @@
 
 
 #if defined(__APPLE__)
-#include <CoreServices/CoreServices.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <stdlib.h>
+#include <string.h>
 
 int macosx_version()
 {
-  SInt32 gestalt_version;
   static int ver = 0; // cached
-  if (0 == ver && (Gestalt(gestaltSystemVersion, &gestalt_version) == noErr)) {
-    ver = gestalt_version;
+  char str[256];
+  size_t strsize = sizeof(str);
+  if (0 == ver && sysctlbyname("kern.osrelease", str, &strsize, NULL, 0) == 0) {
+    // sysctl is major.minor.path
+    *strchr(str, '.') = '\0';
```

Comment:
While rewriting the code, I thought it would even be better to use memchr() to ensure the result is really a string and then use sscanf() to validate the string syntax (ie if it's not X.Y.Z, there must be something fishy).

What do you think of this?

```c
if (memchr(str, '\0', strsize) {
  int major, minor, patch;
  if (sscanf(str, "%d.%d.%d", &major, &minor, &patch) == 3) {
    ver = 0x1000 + major*10;
  }
}
```